CrowsZA/functions/fn_surrenderChance.sqf

```diff
@@ -0,0 +1,137 @@
+/*/////////////////////////////////////////////////
+Author: Landric
+			   
+File: fn_surrenderChance.sqf
+Parameters:
+	unit					- (unit)
+	confrontationRadius		- (float) radius within which pointing a weapon at the unit will trigger a response
+	surrenderChance			- (float) min: 0 max: 1
+	holdFire				- (bool) will unit hold fire until confronted
+	hesitation				- (float) maximum time unit will stall before acting once confronted
+
+Return: bool        		- True if method ran successfully, false if aborted (i.e. bad params were passed)
+
+// This code inspired by CQB Interactions By Flex7103
+// https://steamcommunity.com/sharedfiles/filedetails/?id=2942202773
+
+*///////////////////////////////////////////////
+params [
+	["_unit", objNull, [objNull]],
+	["_confrontationRadius", 15, [15]],
+	["_surrenderChance", 0.33, [0.33]],
+	["_holdFire", true, [true]],
+	["_hesitation", 0, [0]]
+];
+
+
+if(isNull _unit) exitWith { false };
+
+
+// TODO: Add check if unit has already had this module applied/is already surrendered/etc
+
+
+if(_holdFire) then {
+	_unit setUnitCombatMode "BLUE";
+	_unit setCombatBehaviour "CARELESS";
+};
+
+
+private _trigger = createTrigger ["EmptyDetector", getPos _unit, false];
+_trigger setTriggerArea [_confrontationRadius, _confrontationRadius, 0, false];
+_trigger setTriggerActivation ["ANYPLAYER", "PRESENT", true];
+
+_trigger setVariable ["_unit", _unit];
```

Comment:
Is it on purpose that all these variable is only set locally? So these vars will only be possible to be read from the client that apply the module? If you want the varible to be set so all clients can read it, you gotta add a ``true`` at the end, so ``_trigger setVariable ["_unit", _unit, true];`. That signals it should broadcast the variable across network.

Reply:
nwm. I see that this is run only on the zeus machine that inits it. Should probably have a safeguard with a single variable indentifying its already running, to avoid another zeus to apply it on the same unit. Will more than likely cause conflicting behaviour if both scripts tries to run at the same time.